mimic/rest/rackconnect_v3_api.py

```diff
@@ -0,0 +1,447 @@
+        endpoint.
+        """
+        return RackConnectV3Region(
+            iapi=self,
+            uri_prefix=uri_prefix,
+            session_store=session_store,
+            region_name=region,
+            default_lbs=self.default_lbs).app.resource()
+
+
+@attributes(
+    [Attribute("id", default_factory=lambda: text_type(uuid4()),
+               instance_of=text_type),
+     Attribute("name", default_value=u"default", instance_of=text_type),
+     Attribute("port", default_value=80, instance_of=int),
+     Attribute("status", default_value=u"ACTIVE", instance_of=text_type),
+     Attribute("status_detail", default_value=None),
+     Attribute("virtual_ip", default_factory=random_ipv4),
+     Attribute('nodes', default_factory=list, instance_of=list)],
+    apply_with_cmp=False)
+class LoadBalancerPool(object):
+    """
+    Represents a RackConnecvt v3 Load Balancer Pool.
+
+    Load balancer pools cannot be created via the API, they have to just
+    already exist on the account.
+
+    :param text_type id: The ID of the load balancer pool - is randomly
+        generated if not provided
+    :param text_type name: The name of the load balancer pool - defaults to
+        'default'
+    :param int port: The incoming port of the load balancer - defaults to 80
+    :param text_Type status: The status of the load balancer - defaults to
+        "ACTIVE"
+    :param text_type status_detail: Any details about the status - defaults
+        to None
+    :param list nodes: :class:`LoadBalancerPoolNode`s
+    """
+    def as_json(self):
+        """
+        Create a JSON-serializable representation of the contents of this
+        object, which can be used in a REST response for a request for the
+        details of this particular object
+        """
+        # no dictionary comprehensions in py2.6
+        response = dict([
+            (attr.name, getattr(self, attr.name))
+            for attr in LoadBalancerPool.characteristic_attributes
+            if attr.name != "nodes"])
+        response['node_counts'] = {
+            "cloud_servers": len(self.nodes),
+            "external": 0,
+            "total": len(self.nodes)
+        }
+        return response
+
+    def node_by_cloud_server(self, cloud_server_id):
+        """
+        Find a node by its cloud server ID.
+
+        :return: a :class:`LoadBalancerPoolNode` with a server id matching
+            the given cloud server, or `None` if none are found
+        """
+        return next((node for node in self.nodes
+                     if node.cloud_server == cloud_server_id),
+                    None)
+
+    def node_by_id(self, node_id):
+        """
+        Find a node by it's node_id.
+
+        :return: a :class:`LoadBalancerPoolNode` with the given node id,
+            or `None` if none are found
+        """
+        return next((node for node in self.nodes if node.id == node_id), None)
+
+
+# XXX: External nodes are not currently supported in RackConnectV3
```

Comment:
XXX sounds like a TODO. Is there something to be done here? Perhaps this should simply be part of a docstring?